cosmo_tester/resources/agent/winrm-agent-blueprint/plugins/mock-plugin/mock_plugin/tasks.py

```diff
@@ -14,15 +14,56 @@
 #  * limitations under the License.
 
 import os
+import getpass
 
 from cloudify.decorators import operation
+from cloudify.exceptions import NonRecoverableError
 
 
 @operation
-def run(**_):
-    pass
+def prepare_userdata(ctx, service_user, service_password, **_):
+    userdata = """
+#ps1_sysnative
+winrm quickconfig -q
+winrm set winrm/config              '@{MaxTimeoutms="1800000"}'
+winrm set winrm/config/winrs        '@{MaxMemoryPerShellMB="300"}'
+winrm set winrm/config/service      '@{AllowUnencrypted="true"}'
+winrm set winrm/config/service/auth '@{Basic="true"}'
+&netsh advfirewall firewall add rule name="WinRM 5985" protocol=TCP dir=in localport=5985 action=allow
+&netsh advfirewall firewall add rule name="WinRM 5986" protocol=TCP dir=in localport=5986 action=allow
+    """     # noqa
+
+    if service_user:
+        userdata += """
+&net user {user} '{password}' /add
+&net localgroup "Administrators" "{user}" /add
+        """.format(user=service_user, password=service_password)
+
+    ctx.instance.runtime_properties['userdata'] = userdata
 
 
 @operation
-def get_env_variable(env_variable, **_):
-    return os.environ[env_variable]
+def test_app(ctx, service_user, service_password, **_):
+    current_user = getpass.getuser()
+    if not service_user:
```

Comment:
Not critical, but as a point of style I think it's generally better to do:
if <positive>:
  do thing
else:
  do other thing

Rather than starting with if not <thing>

Reply:
You're right. Positive test first... I usually do this, not sure why I didn't here. I'll commit a change.